tests/integration/pilot/vm/legacy_os_vm_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vm
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework/image"
+
+	"istio.io/istio/pkg/test/framework/components/namespace"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+type VMConfig struct {
+	VMHub   string
+	VMTag   string
+	VMImage string
+}
+
+func TestVmTraffic(t *testing.T) {
+	// read from a config list to construct the test matrix
+	file, err := os.Open("testdata/OS_Support_List")
```

Comment:
you can use fileutil.ReadFile

Reply:
Deleted the file and use a list instead.